prov/sm2/src/sm2_av.c

```
@@ -45,154 +46,95 @@ static int sm2_av_close(struct fid *fid)
-							 &util_addr);
-				ofi_mutex_unlock(&util_av->lock);
-			}
-		} else {
-			FI_WARN(&sm2_prov, FI_LOG_AV,
-				"AV insert failed. The maximum number of AV "
-				"entries shm supported has been reached.\n");
-			ret = -FI_ENOMEM;
+		FI_WARN(&sm2_prov, FI_LOG_AV, "fi_av_insert(): About to call sm2_coordinator_allocate_entry() on addr %s, my_pid: %d, count: %zu,  \n", (char*) addr, getpid(), count);
+		ret = sm2_coordinator_allocate_entry(addr, &sm2_av->sm2_mmap, &util_addr, false);
+		FI_WARN(&sm2_prov, FI_LOG_AV, "fi_av_insert(): finished sm2_coordinator_allocate_entry() resulting AV Found = %d \n", util_addr);
+		if (ret && util_av->eq) {
+			ofi_av_write_event(util_av, i, -ret, context);
 		}
+		if (ret) continue;
 
-		FI_INFO(&sm2_prov, FI_LOG_AV, "fi_addr: %" PRIu64 "\n", util_addr);
-		if (fi_addr)
-			fi_addr[i] = util_addr;
-
-		if (ret) {
-			if (util_av->eq)
-				ofi_av_write_event(util_av, i, -ret, context);
-			if (shm_id >= 0)
-				sm2_map_del(sm2_av->sm2_map, shm_id);
-			continue;
+		if (flags & FI_AV_USER_ID) {
+			sm2_av->sm2_aux[util_addr].cqfid = fi_addr[i];
+			FI_INFO(&sm2_prov, FI_LOG_AV, "fi_addr: %d, USER_ID: %" PRIu64 "\n", util_addr, fi_addr[i]);
 		} else {
-			assert(shm_id >= 0 && shm_id < SM2_MAX_PEERS);
-			if (flags & FI_AV_USER_ID) {
-				assert(fi_addr);
-				sm2_av->sm2_map->peers[shm_id].fiaddr = fi_addr[i];
-			} else {
-				sm2_av->sm2_map->peers[shm_id].fiaddr = util_addr;
-			}
-			succ_count++;
-			sm2_av->used++;
+			sm2_av->sm2_aux[util_addr].cqfid = util_addr;
+			FI_INFO(&sm2_prov, FI_LOG_AV, "fi_addr: %d\n", util_addr);
 		}
 
-		assert(sm2_av->sm2_map->num_peers > 0);
-
-		dlist_foreach(&util_av->ep_list, av_entry) {
-			util_ep = container_of(av_entry, struct util_ep, av_entry);
-			sm2_ep = container_of(util_ep, struct sm2_ep, util_ep);
-			sm2_map_to_endpoint(sm2_ep->region, shm_id);
-			sm2_ep->region->max_sar_buf_per_peer =
-				SM2_MAX_PEERS / sm2_av->sm2_map->num_peers;
+		if (fi_addr) {
```

do we silently drop AV_USER_ID  support?

---

I think we might have.

---

I plan to add this back in with a follow up PR specific to this